react-validated-input-component/validated-input/src/ValidatedInput.tsx

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
+
+export default function ValidatedInput() {
+  const [password, setPassword] = useState();
+  const [isValid, setValid] = useState(false);
```

Comment:
Since `isValid` and `validationMessage` can be computed from the `password` state variable, you are better off not using state for them. Just declare them as local variables, set them in `validatePassword()`, and all should go well.